app/models/load_balancer.rb

```diff
@@ -2,13 +2,17 @@ class LoadBalancer < ApplicationRecord
   belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::NetworkManager"
   belongs_to :cloud_tenant
 
+  has_many :load_balancer_health_checks
   has_many :load_balancer_listeners
-  has_many :load_balancer_pools
-  has_many :load_balancer_pool_members, :through => :load_balancer_pools
+  has_many :load_balancer_pools, :through => :load_balancer_listeners
+  has_many :load_balancer_pool_members, -> { distinct }, :through => :load_balancer_pools
 
   has_many :network_ports
   has_many :cloud_subnet_network_ports, :through => :network_ports
   has_many :cloud_subnets, :through => :cloud_subnet_network_ports
   has_many :floating_ips, :through => :network_ports
   has_many :security_groups, -> { distinct }, :through => :network_ports
+
+  has_many :vms, -> { distinct }, :through => :load_balancer_pool_members
+  has_many :resource_groups, -> { distinct }, :through => :load_balancer_pool_members
```

Comment:
This relationship should probably be moved to the Azure-specific `LoadBalancer` class (when there is one).  I don't think any other clouds are using `ResourceGroups` yet.